packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -446,6 +459,17 @@ export class GarbageCollector implements IGarbageCollector {
             && !gcOptions.disableGC
         );
 
+        const currentVersion = pkgVersion;
+        const oldestTrackGCBlobStateVersion = this.mc.config.getString(trackGCBlobStateVersionKey);
+        const versionIsCompatible =
+            oldestTrackGCBlobStateVersion === undefined ||
+            currentVersionGreaterOrEqualToVersion(currentVersion, oldestTrackGCBlobStateVersion);
+        const trackGCBlobState = this.mc.config.getBoolean(trackGCBlobStateKey) === true;
```

Comment:
Do we need a separate key to enable this? Or should be just have an option to disable it (which can be done via the versions key)?

Reply:
What version key would we want to use if we want to keep this disabled? Like 1.0.0? And what about rollback and tests?

Reply:
Not sure and it's not a big deal. I was just wondering if we can avoid it. Like set it to some ridiculous version 100.0.0 or something so that no client matches it until we fix the bug and update it. Just a thought.

Reply:
Actually now that I am thinking about it, we should ship dark and enable via feature flag. This change is risky and we should rollout in stages and keep an eye out.
How will we know if something goes wrong and the feature is not working as expected?